### PR TITLE
Fix pointer serialization

### PR DESCRIPTION
**Pull request by roman-khimov**
### Problem

#2815 is caused by the pointer element that was omitted during serialization and then deserialization code saw things it's not supposed to see.

### Solution

Handle pointers.

### CODE REVIEW

File: cli/vm/cli.go [hunk 1]
```diff
@@ -640,12 +647,22 @@ func handleSlots(c *cli.Context) error {
...
+		err = resetState(c.App, tx, uint32(height))
+	} else {
+		err = resetState(c.App, tx)
 	}
-
-	return resetState(c.App, tx)
+	if err != nil {
+		return err
+	}
+	if c.IsSet(gasFlagFullName) {
```
Comment (AnnaShaleva) [inline cli/vm/cli.go:660]: We have a special `handleLoadDeployed` command, it does magic with `ReuseVM`, so we need to add the same code to the end of `handleLoadDeployed`.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

File: cli/vm/cli.go (hunk 1 above)
Comment (roman-khimov) [reply cli/vm/cli.go:663]: Fixed. I've also changed the default behavior of `loadtx` to something probably more expected.

File: cli/vm/cli.go [hunk 2]
```diff
@@ -829,6 +832,9 @@ func handleLoadTx(c *cli.Context) error {
 		return err
 	}
 	v := getVMFromContext(c.App)
+	if v.GasLimit == -1 {
+		v.GasLimit = tx.SystemFee
+	}
```
Comment (AnnaShaleva) [inline cli/vm/cli.go:837]: A good one, I didn't think about it.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**